tests/integrations/python/tests/utils/common.py

```
@@ -2615,4 +2615,425 @@ def assert_valid_input_tokens_response(response: Any, library: str):
+    
+    elif expected_type == "url_citation":
+        # url_citation: url, title, start_index, end_index
+        if hasattr(annotation, "url"):
+            assert isinstance(annotation.url, str), "url should be a string"
+        if hasattr(annotation, "title"):
+            assert isinstance(annotation.title, str), "title should be a string"
+        if hasattr(annotation, "start_index") and hasattr(annotation, "end_index"):
+            assert isinstance(annotation.start_index, int), "start_index should be an integer"
+            assert isinstance(annotation.end_index, int), "end_index should be an integer"
+            assert annotation.end_index > annotation.start_index, \
+                f"end_index should be > start_index"
```

Ruff (0.14.11)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)